src/app/shared/menu/date/standard-date-range.component.ts

```diff
@@ -1,47 +1,52 @@
 import { Component, Input, Output, EventEmitter, OnChanges } from '@angular/core';
-import { IntervalModel, INTERVAL_HOURLY, INTERVAL_DAILY, INTERVAL_WEEKLY, INTERVAL_MONTHLY } from '../../../ngrx';
-import { isMoreThanXDays, endOfTodayUTC,
-  beginningOfThreeMonthsUTC, beginningOfThisYearUTC,
-  TODAY, THIS_WEEK, TWO_WEEKS, THIS_MONTH, THREE_MONTHS, THIS_YEAR,
-  YESTERDAY, LAST_WEEK, PRIOR_TWO_WEEKS, LAST_MONTH, PRIOR_THREE_MONTHS, LAST_YEAR } from '../../util/date.util';
+import { IntervalModel, INTERVAL_HOURLY, INTERVAL_MONTHLY } from '../../../ngrx';
+import { dayOfWeekDateFormat, dayMonthDateFormat, monthDateYearFormat,
+  beginningOfThisWeekUTC, beginningOfLastWeekUTC, endOfLastWeekUTC, beginningOfLast7DaysUTC, beginningOfThisWeekPlus7DaysUTC,
+  beginningOfThisMonthUTC, beginningOfLastMonthUTC, endOfLastMonthUTC, beginningOfLast28DaysUTC, beginningOfLast30DaysUTC,
+  beginningOfThisMonthPlusTwoMonthsUTC, beginningOfLast90DaysUTC, beginningOfThisYearUTC, beginningOfLast365DaysUTC,
+  THIS_WEEK, LAST_WEEK, LAST_7_DAYS, THIS_WEEK_PLUS_7_DAYS,
+  THIS_MONTH, LAST_MONTH, LAST_28_DAYS, LAST_30_DAYS,
+  THIS_MONTH_PLUS_2_MONTHS, LAST_90_DAYS, THIS_YEAR, LAST_365_DAYS, OTHER } from '../../util/date.util';
 
 @Component({
   selector: 'metrics-standard-date-range',
   template: `
-    <prx-select single="true" [options]="rangeOptions" [selected]="standardRange" (onSelect)="onStandardRangeChange($event)"></prx-select>
-  `
+    <ul *ngFor="let group of rangeOptions" class="group">
+      <li *ngFor="let range of group">
+        <button class="btn-link" (click)="onStandardRangeChange(range)">
+          {{range}} <span>{{ getRangeDesc(range) }}</span>
+        </button>
+      </li>
+    </ul>
+  `,
+  styleUrls: ['./standard-date-range.component.css']
 })
 export class StandardDateRangeComponent implements OnChanges {
   @Input() standardRange: string;
   @Input() interval: IntervalModel;
   @Output() standardRangeChange = new EventEmitter<string>();
-  rangeOptions: string[] = [];
+  rangeOptions: string[][] = [];
 
   ngOnChanges() {
     this.genRanges();
   }
 
   genRanges() {
-    this.rangeOptions = [TODAY, THIS_WEEK, TWO_WEEKS, THIS_MONTH];
-
-    if (this.interval === INTERVAL_DAILY || this.interval === INTERVAL_WEEKLY || this.interval === INTERVAL_MONTHLY ||
-      (this.interval === INTERVAL_HOURLY && !isMoreThanXDays(40, beginningOfThreeMonthsUTC(), endOfTodayUTC()))) {
-      this.rangeOptions.push(THREE_MONTHS);
-    }
-
-    if (this.interval === INTERVAL_DAILY || this.interval === INTERVAL_WEEKLY || this.interval === INTERVAL_MONTHLY ||
-      (this.interval === INTERVAL_HOURLY && !isMoreThanXDays(40, beginningOfThisYearUTC(), endOfTodayUTC()))) {
-      this.rangeOptions.push(THIS_YEAR);
+    if (this.interval !== INTERVAL_MONTHLY) {
```

Comment:
A bit nitpicky, but I find this kind of hard to read.  Could it be reorganized more like a switch statement?  (One block for every possible `this.interval`)

Reply:
And yeah, as I continue reading, `getRangeDesc()` is much more intuitive than this one.